Docker CE support for hpc-rocky-linux-8 image

m4r1k opened the pull request:
**Summary**
This PR restores support for Docker CE when Slurm is deployed on the `hpc-rocky-linux-8` image.

**Problem**
Currently, setting `docker.enabled: true` causes deployments to fail on the official Google Cloud HPC image (Rocky Linux 8).
*   The existing installation method relies on the `get.docker.com` script.
*   As documented in [docker-install#373](https://github.com/docker/docker-install/issues/373), this script does not support Rocky Linux.

**Motivation**
Docker is essential for several HPC workflows, including [Nextflow](https://www.nextflow.io/docs/latest/index.html). While Nextflow supports other runtimes, Docker simplifies dependency management by keeping environments self-contained within [Container Images](https://www.nextflow.io/docs/latest/container.html).

**Verification**
I have attached the blueprint used to validate the Docker deployment.
*   [blueprint.yaml](https://github.com/user-attachments/files/23679453/blueprint.yaml)

bytetwin commented on modules/scripts/startup-script/files/install_docker.yml:
```diff
@@ -64,25 +113,33 @@
+    - Restart Docker
+
+  - when:
+    - enable_docker_world_writable
```
`when: enable_docker_world_writable` as its only a single boolean check

m4r1k commented on modules/scripts/startup-script/files/install_docker.yml:
```diff
@@ -64,25 +113,33 @@
+        content: |
+          [Socket]
+          SocketMode=0666
+      when: enable_docker_world_writable
```
Good catch. Ack!

m4r1k commented:
In the latest patch I've addressed all comments. Here the playbook run logs:
```
PLAY [Install and configure Docker] ********************************************

TASK [Gathering Facts] *********************************************************
ok: [localhost]

TASK [Gather distribution facts] ***********************************************
ok: [localhost]

TASK [Check if docker is installed] ********************************************
ok: [localhost]

TASK [Add Docker GPG key] ******************************************************
changed: [localhost]

TASK [Add Docker repository for Rocky Linux] ***********************************
changed: [localhost]

TASK [Install Docker Engine Community Edition] *********************************
changed: [localhost]

TASK [Download Docker Installer] ***********************************************
skipping: [localhost]

TASK [Install Docker Engine Community Edition] *********************************
skipping: [localhost]

TASK [Create Docker daemon configuration] **************************************
skipping: [localhost]

TASK [Create Docker service override directory] ********************************
changed: [localhost]

TASK [Create Docker service override configuration] ****************************
changed: [localhost]

TASK [Create Docker socket override directory] *********************************
changed: [localhost]

TASK [Create Docker socket override configuration] *****************************
changed: [localhost]

TASK [Delete Docker socket override configuration] *****************************
skipping: [localhost]

RUNNING HANDLER [Reload SystemD] ***********************************************
ok: [localhost]

RUNNING HANDLER [Recreate Docker socket] ***************************************
changed: [localhost]

TASK [Retart Docker] ***********************************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=13   changed=9    unreachable=0    failed=0    skipped=4    rescued=0    ignored=0
```

Of course, Docker works like a charm:
```
srun -p c4highmem8 -N 1 docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
17eec7bbc9d7: Pulling fs layer
ea52d2000f90: Download complete
17eec7bbc9d7: Download complete
17eec7bbc9d7: Pull complete
Digest: sha256:d4aaab6242e0cace87e2ec17a2ed3d779d18fbfd03042ea58f2995626396a274
Status: Downloaded newer image for hello-world:latest

Hello from Docker!
```

bytetwin commented:
/gcbrun
